engine/src/main/java/io/camunda/zeebe/engine/state/immutable/TimerInstanceState.java

```diff
@@ -7,17 +7,20 @@
  */
 package io.camunda.zeebe.engine.state.immutable;
 
+import io.camunda.zeebe.engine.processing.scheduled.DueDateTriggerState;
 import io.camunda.zeebe.engine.state.instance.TimerInstance;
 import java.util.function.Consumer;
 
-public interface TimerInstanceState {
+public interface TimerInstanceState extends DueDateTriggerState<TimerInstance> {
 
   /**
    * Finds timers with due date before {@code timestamp}, and presents them to the {@code consumer}
    *
    * @return due date of the next scheduled timer (or {@code -1} if no succeeding timer exists)
    */
-  long processTimersWithDueDateBefore(long timestamp, TimerVisitor consumer);
+  default long processTimersWithDueDateBefore(final long timestamp, final TimerVisitor consumer) {
```

Comment:
:question: Do we need the default impl? Here, we will not persist the cache, right
?

Reply:
The default is just so I don't have to edit every call yet :smile: We'll have to remove it long term of course